DEVELOPMENT.md

```diff
@@ -87,6 +87,16 @@ Maintain the same quality for production and test code.
 
 When writing a Git commit message, follow these [guidelines](https://chris.beams.io/posts/git-commit/).
 
+### Git merge strategy
+
+We prefer to `rebase and merge`, so cleaning up and squashing commits on the feature branch is best.
```

Comment:
While we are at it would it be helpful to point to something like https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html too so that the next paragraph makes sense. (It talks about keeping commits structured and changes not going back and forth within commits).

Reply:
What link would you prefer? https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges This one seems like a pretty good starting point.

@hashhar I think the doc you linked isn't so much an explanation of squash & merging as it is a tip on specific git tools (fixup and autosquash). For me, I typically use git commit --amend when fixing up the previous commit. Maybe a more generic article like this: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History would be better? Let me know your thoughts!

Reply:
I mentioned fixup and autosquash mainly because in actual practice most of the times people run into a state where the commit they want to edit isn't the previous commit - it's something older.

e.g. 3 commits A, B and C. Reviewers asks to change something in A. You add that as a fixup commit to A so that in the end you can autosquash and git will handle which commits go where during the rebase.

However - now thinking about it I feel we can leave this detail out - it's more likely to confuse new people and if anyone or any PR needs this then the reviewers can point people in the direction.